feasibility-triangle/aktin-client/client-exec-scripts/call-cql.sh

```diff
@@ -10,9 +10,24 @@ echo "----END REQUEST----" >> aktin-requests.log
 
 RESP=$(sh execute-cql.sh "$QUERY_INPUT")
 
-if [ $CLIENT_OBFUSCATE = true ]; then
-  if [ $RESP != 0 ];then
-    RESP=$(($RESP - ($RESP % 10) + 10))
+echo $RESP >> aktin-requests.log
+
+getObfuscatedResp() {
+    OBFUSCATION_INTEGER=$(grep -m1 -ao '[0-5]' /dev/random | head -n1)
+    OBFUSCATION_SIGN=$(grep -m1 -ao '[-+]' /dev/random)
+
+    if [ $OBFUSCATION_INTEGER -eq 0 ] && [ $OBFUSCATION_SIGN == '-' ]; then
+        getObfuscatedResp $1
+    else
+       echo $(($1 $OBFUSCATION_SIGN $OBFUSCATION_INTEGER))
+    fi
+
+}
```

Comment:
IMHO 2 requests to a random generator followed by a potential recursion are not necessary in this case.

You can as well:
1. read a certain amount of bytes from `/dev/random` or `/dev/urandom`
2. convert them to signed integers
3. create a range using the modulo operator

Something along the lines of:

```shell
random_number=$(($(dd if=/dev/urandom bs=1 count=1 2>/dev/null | od -A n -t "d1") % 6))
```
This will yield a random number in the desired range of [-5, 5].

You can adjust byte sizes as you see fit. Since this converts to signed integers, there are more negative values than positive ones. If this is a problem, then feel free convert to unsigned integers and use an offset instead:

```shell
random_number=$(($(dd if=/dev/urandom bs=1 count=1 2>/dev/null | od -A n -t "u1") % 11 - 5))
```

### Additional information regarding `/dev/random` vs `/dev/urandom`

From the official documentation:
> The /dev/random device is a legacy interface which dates back to
       a time where the cryptographic primitives used in the
       implementation of /dev/urandom were not widely trusted.  It will
       return random bytes only within the estimated number of bits of
       fresh noise in the entropy pool, blocking if necessary.
       /dev/random is suitable for applications that need high quality
       randomness, and can afford indeterminate delays.

> When read, the /dev/urandom device returns random bytes using a
       pseudorandom number generator seeded from the entropy pool.
       Reads from this device do not block (i.e., the CPU is not
       yielded), but can incur an appreciable delay when requesting
       large amounts of data.

> When read during early boot time, /dev/urandom may return data
   prior to the entropy pool being initialized.

> The /dev/random interface is considered a legacy interface, and
       /dev/urandom is preferred and sufficient in all use cases, with
       the exception of applications which require randomness during
       early boot time;

You have to decide whether it is tolerable to have a potential blocking behavior. Other than that `/dev/urandom` seems to be a good fit and should be preferred. However, behavior within a Docker container should be investigated with regards to the seed file.

Reply:
The singed, modulo 6 variant gives you double the amount the zero:
```
for ((i=0;$i<10000;i=$i+1)) do echo "$(( $(od -An -N1 -td1 /dev/urandom) % 6 ))"; done | sort -n | uniq -c
```

The unsigned, modulo 11 minus 5 variant works:
```
for ((i=0;$i<10000;i=$i+1)) do echo "$(( $(od -An -N1 -tu1 /dev/urandom) % 11 - 5 ))"; done | sort -n | uniq -c
```

Using `dd` is not necessary, because we can request `od` to read only one byte by supplying `-N1`.

Reply:
Thanks for double-checking. Thus, the version using signed conversion is ruled out.

In terms of readability and ease of use, I'd go with your proposed version using `$RANDOM` anyway.